### PR TITLE
misc/mdns: Make MdnsService::new sync by using std::net::UdpSocket::bind

### PR DESCRIPTION
**Pull request by mxinden**
MdnsService uses an IP address to create a UDP socket. The address does
not need to be resolved. Therefore one can use std's UdpSocket::bind
instead of the async counterpart from async-std. As a result
MdnsService::new and MdnsService::silent don't need to be async.

https://github.com/libp2p/rust-libp2p/pull/1328#discussion_r360669993

(//CC @DemiMarie-parity)

### CODE REVIEW

File: misc/mdns/src/behaviour.rs [hunk 1]
```diff
@@ -90,7 +90,7 @@ impl<TSubstream> Mdns<TSubstream> {
     /// Builds a new `Mdns` behaviour.
     pub async fn new() -> io::Result<Mdns<TSubstream>> {
```
Comment (twittner) [inline misc/mdns/src/behaviour.rs]:
```suggestion
    pub fn new() -> io::Result<Mdns<TSubstream>> {
```
?

**Comment by DemiMarie-parity** [comment]
Looks good to me, other than @twittner's correction.

File: misc/mdns/src/behaviour.rs (hunk 1 above)
Comment (mxinden) [reply misc/mdns/src/behaviour.rs]: Right. That makes a lot of sense. Thanks.

Comment (mxinden) [reply misc/mdns/src/behaviour.rs]: Right. That makes a lot of sense. Thanks.

**Comment by mxinden** [comment]
Should we merge this after we released `v0.14.0`, given that it is a breaking change to the API surface?

**Comment by tomaka** [comment]
> Should we merge this after we released v0.14.0, given that it is a breaking change to the API surface?

We've already got a few breaking changes in, so let's go ahead with that.

**Pull request merged**